libuavcan/include/uavcan/build_config.hpp

```diff
@@ -220,11 +221,20 @@ namespace uavcan
 static const unsigned MemPoolBlockSize = UAVCAN_MEM_POOL_BLOCK_SIZE;
 #elif defined(__BIGGEST_ALIGNMENT__) && (__BIGGEST_ALIGNMENT__ <= 8)
 /// Convenient default for GCC-like compilers - if alignment allows, pool block size can be safely reduced.
+#ifdef UAVCAN_USE_FD
+static const unsigned MemPoolBlockSize = 64;
```

Comment:
64 bytes are not sufficient for CAN FD - we also need to take into account the CAN ID field and other overheads.

My quick estimation shows that we need (8 bytes pointer + 64 bytes payload + 4 bytes CAN ID + 1 byte DLC + 3 bytes padding + 8 bytes TX deadline + 1 byte QoS + 1 byte IO flags + 6 bytes padding) = 96 bytes large blocks, regardless of the platform. Huh, @davids5 is going to love this.

Reply:
I kept meaning to ask you about how you calculate this. Can I put your math here into a comment? It'd be great to understand where the initial values came from.

Also: when you say "8 bytes pointer" do you mean two 32-bit pointers or a single 64-bit?

Also, Also: The unit tests ran with this value even under valgrind. How can we write a test that validates this value on a given platform?

Reply:
> Can I put your math here into a comment?

Sure.

> It'd be great to understand where the initial values came from.

The largest dynamically allocated object type is `uavcan::CanTxQueue::Entry`, so I just pulled the numbers from there.

> Also: when you say "8 bytes pointer" do you mean two 32-bit pointers or a single 64-bit?

No, I mean a single 64-bit pointer, this is the worst case. On a 32-bit system, it would be 32-bit, but the alignment reasons require us to keep the block size aligned at 8 bytes at least (maybe more depending on the platform; 96-byte large blocks can be aligned at 32 bytes, which should suit every sensible use case).

> The unit tests ran with this value even under valgrind. How can we write a test that validates this value on a given platform?

*What no that can't be.* The allocator is designed to fail if you're trying to allocate more bytes than the size of the memory block: https://github.com/UAVCAN/libuavcan/blob/c044630390370d800a22e7746cd949ddd3c59bb1/libuavcan/include/uavcan/dynamic_memory.hpp#L162-L165